src/components/navigation/z-navigation-tabs/index.tsx

```diff
@@ -0,0 +1,127 @@
+import { Component, Prop, h, Listen, Element, State, Watch, Host } from '@stencil/core';
+
+import {
+  TabSizeBean,
+  TabSizeEnum,
+  TabOrientationBean,
+  TabOrientationEnum
+} from '../../../beans';
+import { ZNavigationTab } from '../z-navigation-tab';
+
+@Component({
+  tag: 'z-navigation-tabs',
+  styleUrl: 'styles.css',
+  shadow: true
+})
+export class ZNavigationTabs {
+  @Element() host: HTMLElement;
+
+  /** Available orientation: `horizontal` and `vertical`. Defaults to `horizontal`. */
+  @Prop({ reflect: true }) orientation?: TabOrientationBean = TabOrientationEnum.horizontal;
+  /** Available sizes: `big` and `small`. Defaults to `big`. */
+  @Prop({ reflect: true }) size?: TabSizeBean = TabSizeEnum.big;
+  /** State for the navigation */
+  @State() canNavigate: boolean;
+  @State() canNavigatePrev: boolean;
+  @State() canNavigateNext: boolean;
+
+  tabsNav: HTMLElement;
+
+  /** Set all childrens (tab) orientation attribute : horizontal (default) or vertical */
+  @Watch('orientation')
+  setChildrenOrientation() {
+    const children = this.host.children;
+    for (let i = 0; i < children.length; i++) {
+      children[i].setAttribute('orientation', this.orientation);
+    }
+  }
+
+  /** When resize check if the navigation buttons are needed */
+  @Listen('resize', { target: 'window', passive: true })
+  checkScrollVisible() {
```

Comment:
Since you use the `orientation` prop to change the behavoiur of the component more the one time, you can add state vars: 
```typescript
  @State() direction : 'Top' | 'Left' = this.orientation == TabOrientationEnum.vertical ? 'Top' : 'Left';     
  @State() dimension : 'Height' | 'Width' = this.orientation == TabOrientationEnum.vertical ? 'Height' : 'Width';     
```
and rewrite this snippet like this: 
```typescript
    this.canNavigate = this.tabsNav[`scroll${this.dimension}`] > this.tabsNav[`client${this.dimension}`];
```